js/queries.js

```diff
@@ -54,6 +58,15 @@ function getBreakpoint(state, location) {
   return getBreakpoints(state).get(makeLocationId(location));
 }
 
+function getBreakpointByActor(state) {
+  return getBreakpoints(state).valueSeq()
+    .groupBy(bp => bp.getIn(["location", "actor"]));
+}
+
+function getBreakpointsForSource(state, actor) {
+  return getBreakpointByActor(state).get(actor);
+}
```

Comment:
this is definitely going to conflict with your breakpoint work, but hopefully it's very similar to what you have